### ISSUE TITLE
Warning: Reading certificate from stdin since no -in or -new option is given

### ISSUE THREAD

**Issue by boris-kolpackov** [build2/build2#353]
The following command line (executed as part of `bpkg-fetch`) now issues a warning:

```
fetching pkg:cppget.org/beta (complements pkg:cppget.org/alpha)
...
openssl x509 -sha256 -noout -fingerprint
Warning: Reading certificate from stdin since no -in or -new option is given
```

This was original observed on Mac OS (which by default uses LibreSSL) but with what appears to be OpenSSL installed from Homebrew:

```
$ openssl --version
OpenSSL 3.2.0 23 Nov 2023 (Library: OpenSSL 3.2.0 23 Nov 2023)

$ which openssl
/opt/homebrew/bin/openssl
```

**Comment by helmesjo** [comment]
Happened now on latest staged on WSL as well:

```
openssl x509 -sha256 -noout -fingerprint
Warning: Reading certificate from stdin since no -in or -new option is given
```
`OpenSSL 3.2.0 23 Nov 2023 (Library: OpenSSL 3.2.0 23 Nov 2023)`

```
bdep --version
bdep 0.17.0-a.0.260c9251aa08
libbpkg 0.17.0-a.0.ca2934d78b84
libbutl 0.17.0-a.0.d249dff880ed
Copyright (c) 2014-2023 the build2 authors.
This is free software released under the MIT license.
```

**Comment by boris-kolpackov** [comment]
This is not easy to fix: there is no way to specify `stdin` explicitly with `-in` (e.g., the customary `-in -` does not work) nor is there are a way to suppress the warning (e.g., with a quiet option or some such). But we are not the only ones affected by this and it looks like the OpenSSL folks are planning to fix this on their end: https://github.com/openssl/openssl/issues/22893

**Comment by GavinFrazar** [comment]
hey, I'm just a passerby who found this issue through the openssl issue you linked.

Workaround I use to suppress the warning is to pass `-in /dev/stdin`.

**Comment by boris-kolpackov** [comment]
Worked around for Unix in https://github.com/build2/bpkg/commit/40bb034f57cf37c1cbdcef337733507c230c6004 using the `/dev/stdin` hack. Upstream fixed this in OpenSSL 3.3.0 and it's unlikely we will bother with workarounds for Windows. So I am going to close this issue.

**Comment by karen-arutyunov** [comment]
Upstream has also fixed this in OpenSSL 3.2.2 as well, as per https://github.com/openssl/openssl/issues/22893#issuecomment-2095415586.
